### PR TITLE
Refactor/python docstrings for regsd estimator

### PR DESCRIPTION
**Pull request by Dioprz**
### Description

Implementation of https://github.com/Crypto-TII/CryptographicEstimators/pull/169 for RegSDEstimator.

### Review process

- A quick glance at the new docstrings would be great.
- Please search by FIX and TODO in the diff view, in case I has moved some previously existent comment.

**Estimator-specific doctests**
```shell
make docker-run
pytest --doctest-modules -n auto -vv cryptographic_estimators/RegSDEstimator
```

**Cumulative test (with all the already migrated docstrings)**
```shell
make docker-pytest-doctests
```

### Pre-approval checklist
- [x] The code builds clean without any errors or warnings
- [x] I've added/updated tests
- [x] I've added/updated documentation if needed

### CODE REVIEW

**Review by FloydZ — CHANGES_REQUESTED** [review]
Looks good to me. But I checked the new generated documentation. And it does not look correct.

Screenshot_20240930_102834

File: cryptographic_estimators/RegSDEstimator/RegSDAlgorithms/regisd_enum.py [hunk 1]
```diff
@@ -23,64 +23,55 @@
...
+        Examples:
+            >>> from cryptographic_estimators.RegSDEstimator.RegSDAlgorithms import RegularISDEnum
+            >>> from cryptographic_estimators.RegSDEstimator import RegSDProblem
+            >>> A = RegularISDEnum(RegSDProblem(n=100,k=50,w=10))
+            >>> A.p()
             4
         """
         return self._get_optimal_parameter("p")
 
+    #TODO: Update examples
```
[the remark this answers is not in the document]
Comment (Dioprz) [reply cryptographic_estimators/RegSDEstimator/RegSDAlgorithms/regisd_enum.py]: Applied

File: cryptographic_estimators/RegSDEstimator/RegSDAlgorithms/regisd_enum.py [hunk 2]
```diff
@@ -23,64 +23,55 @@
...
+            >>> A
+            RegularISD-Enum estimator for the RegSDProblem with parameters (n, k, w) = (100, 50, 10)
+        """
         super(RegularISDEnum, self).__init__(problem, **kwargs)
         n, k, w = self.problem.get_parameters()
         self._name = "RegularISD-Enum"
         self.set_parameter_ranges("p", 0, 30)
         self.set_parameter_ranges("ell", 0, n-k)
+
+    #TODO: Update examples
```
[the remark this answers is not in the document]
Comment (Dioprz) [reply cryptographic_estimators/RegSDEstimator/RegSDAlgorithms/regisd_enum.py]: Done

**Comment by Dioprz** [comment]
Look at how are generated the docstrings now:

image

Test with `make docker-doc`

All the required changes for this were added at https://github.com/Crypto-TII/CryptographicEstimators/pull/191/commits/932828b87e3e2f6ec3fc26733590573bae9b02a3.

Basically I made two changes:
- Include the Napoleon extension.
- I created a custom Sphinx extension able to parse the `**kwargs` parameter, as I used this pattern in all the new docstrings-related PRs (IIRC). Now they look pretty much the same as in the docstrings:

```python
    def __init__(self, n: int, k: int, w: int, memory_bound=inf, **kwargs):
        """Construct an instance of RegSDEstimator.

        Args:
            n (int): Code length.
            k (int): Code dimension.
            w (int): Error weight.
            memory_bound: Memory bound.
            **kwargs: Additional keyword arguments.
                excluded_algorithms: A list/tuple of excluded algorithms (default: None).
                nsolutions: No. of solutions.
        """
```

If you prefer to have this changes in another PR, let me know.

**Help required**

As you can see, I put the `kwargs_formatter.py` directly in the root of the project. I don't know if you prefer to create another directory, or put it directly into the `scripts/` or `docs/` directory so... Just tell me where you want it and I will apply the change. @FloydZ @Javierverbel

**Review by Javierverbel — APPROVED** [review]

**Review by FloydZ — APPROVED** [review]
Looks good. We delete all docs, and the final PR adds the newly generated docs then?

**Comment by Javierverbel** [comment]
yes.

**Pull request merged**